enhancements/console/customize-project-roles.md

```diff
@@ -0,0 +1,167 @@
+---
+title: customize-project-roles
+authors:
+  - "@jerolimov"
+reviewers:
+  - "@spadgett"
+  - "@cvogt"
+approvers:
+  - "@spadgett"
+  - "@cvogt"
+creation-date: 2021-02-25
+last-updated: 2021-03-05
+status: implementable
+---
+
+# Customize Project Access Roles
+
+## Release Signoff Checklist
+
+- [x] Enhancement is `implementable`
+- [x] Design details are appropriately documented from clear requirements
+- [x] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Motivation
+
+Customers with customized roles want to provide easy access for their project admins to restrict access within a project.
+
+OpenShift Console 3.11 allowed customers to customize the list of roles visible in the membership page. This enhancement will provide equivalent functionality in 4.x.
+
+## Summary
+
+Cluster admins want give project admins only access to a customized list of predefined roles to force compliance rules or provide more options.
+
+The OpenShift Developer Console provides a simple way to share access to the currently selected project. This change will only affect the project page in the developer perspective. Users can be assigned here to specific roles on the project access tab. Currently only the roles "Admin", "Edit" and "View" are selectable.
+
+![Project access tab with role select field to select an access role](customize-project-roles-project-access-tab.png)
+
+### Goals
+
+- Add configuration options to allow cluster admins to customize the project access roles.
+- Provide a list of default values to the admin.
+
+### Non-Goals
+
+- Introducing a UI to configure this project access roles without a YAML editor.
+
+## Proposal
+
+### Extending the Console CRD
+
+Similar to [Customize Developer Catalog Categories](./catalog-categories.md) we extend the existing `operator.openshift.io/v1` / `Console` CRD. It provides already a `spec.customization` area where we can add a list of roles.
+
+The list of roles should link to `rbac.authorization.k8s.io/v1` / `Role` or `ClusterRole`.
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Console
+metadata:
+  name: cluster
+  ...
+spec:
+  customization:
+    brand: online
+    developerPerspective:
+      projectAccessRoles:
```

Comment:
I'm not a fan of isolating the config under a `developerPerspective` stanza just because I don't want isolate a feature to one perspective.